packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -272,9 +272,12 @@ function resetComponentStateWhenRemoved(vm: VM) {
     const { state } = vm;
 
     if (state !== VMState.disconnected) {
-        const { tro } = vm;
+        const { tro, component } = vm;
         // Making sure that any observing record will not trigger the rehydrated on this vm
         tro.reset();
+        if (lwcRuntimeFlags.ENABLE_EXPERIMENTAL_SIGNALS) {
+            unsubscribeFromSignals(component);
+        }
```

Comment:
Since we need to do this for every `reset`, can we have a central function to handle it? Something like `resetAndUnsubscribeVM(vm)`.

Reply:
add a new [`resetTemplateObserverAndUnsubscribe`](https://github.com/salesforce/lwc/blob/3c78f17484831774b00c494ab1e009a282f2a175/packages/%40lwc/engine-core/src/framework/component.ts#L93-L100) function!